src/analyze/analyze.c

```diff
@@ -1139,22 +1154,51 @@ static int graph_one(sd_bus *bus, const UnitInfo *u, char *patterns[], char *fro
+                r = graph_one_property(bus, u,
+                                       unit_dependency_to_string(UNIT_AFTER),
+                                       dependency_color[UNIT_AFTER],
+                                       dependency_style[UNIT_AFTER],
+                                       patterns, from_patterns, to_patterns);
                 if (r < 0)
                         return r;
         }
 
         if (IN_SET(arg_dot, DEP_REQUIRE, DEP_ALL)) {
-                r = graph_one_property(bus, u, "Requires", "black", patterns, from_patterns, to_patterns);
-                if (r < 0)
-                        return r;
-                r = graph_one_property(bus, u, "Requisite", "darkblue", patterns, from_patterns, to_patterns);
+                for (int dependency = UNIT_REQUIRES; dependency <= UNIT_PART_OF; ++dependency) {
+                        r = graph_one_property(bus, u,
+                                               unit_dependency_to_string(dependency),
+                                               dependency_color[dependency],
+                                               dependency_style[dependency],
+                                               patterns, from_patterns, to_patterns);
+                        if (r < 0)
+                                return r;
+                }
+
+                r = graph_one_property(bus, u,
+                                       unit_dependency_to_string(UNIT_CONFLICTS),
+                                       dependency_color[UNIT_CONFLICTS],
+                                       dependency_style[UNIT_CONFLICTS],
+                                       patterns, from_patterns, to_patterns);
                 if (r < 0)
                         return r;
-                r = graph_one_property(bus, u, "Wants", "grey66", patterns, from_patterns, to_patterns);
+        }
+
+        if (IN_SET(arg_dot, DEP_FAILURE, DEP_ALL)) {
+                r = graph_one_property(bus, u,
+                                       unit_dependency_to_string(UNIT_ON_FAILURE),
+                                       dependency_color[UNIT_ON_FAILURE],
+                                       dependency_style[UNIT_ON_FAILURE],
+                                       patterns, from_patterns, to_patterns);
                 if (r < 0)
                         return r;
-                r = graph_one_property(bus, u, "Conflicts", "red", patterns, from_patterns, to_patterns);
+        }
+
+        if (IN_SET(arg_dot, DEP_TRIGGERS, DEP_ALL)) {
+                r = graph_one_property(bus, u,
+                                       unit_dependency_to_string(UNIT_TRIGGERS),
+                                       dependency_color[UNIT_TRIGGERS],
+                                       dependency_style[UNIT_TRIGGERS],
+                                       patterns, from_patterns, to_patterns);
                 if (r < 0)
                         return r;
```

Comment:
i think these funciton should have a static const struct array that lists under which arg_dot values to process which dep with what. That would make this function a lot more readable and remove repetitive code.

Reply:
yes, done